### PR TITLE
Admin user index #58

### LINKED ISSUE #57: User Story 56, Admin User Profile Page
```
As an admin user
When I visit a user's profile page ("/admin/users/5")
I see the same information the user would see themselves
I do not see a link to edit their profile
```

DO THESE STORIES FIRST:
- User Story 5, Admin Navigation
- User Story 55, Admin User Index Page

### Checklist:

- [ ] base tests are written
- [ ] base code is written to pass tests
- [ ] edge case ("sad path") tests written (if required)
- [ ] edge case code written (if required)
- [ ] code is reviewed
- [ ] branch is merged
- [ ] all tests still pass

child of #59

child of #125

depends on #119

depends on #58

### LINKED ISSUE #58: User Story 55, Admin User Index Page
```
As an admin user
When I click a new "Users" link in the nav (only visible to admins)
Then my current URI route is "/admin/users"
Only admin users can reach this path.
I see all users in the system who are not merchants nor admins.
Each user's name is a link to a show page for that user ("/admin/users/5")
Next to each user's name is the date they registered
Next to each user's name is a button that says 'Upgrade to Merchant'
```

DO THESE STORIES FIRST:
- User Story 5, Admin Navigation

### Checklist:

- [ ] base tests are written
- [ ] base code is written to pass tests
- [ ] edge case ("sad path") tests written (if required)
- [ ] edge case code written (if required)
- [ ] code is reviewed
- [ ] branch is merged
- [ ] all tests still pass

child of #59

child of #125

depends on #119

### PR DESCRIPTION
**Pull request by Vjp888**
Adds the ability for an admin to view a list of users and individual show page of each user.

All tests passing

closes #58 
closes #57

### CODE REVIEW

File: app/views/shared/_nav.html.erb [hunk 1]
```diff
@@ -2,5 +2,8 @@
 <%= link_to 'Home', root_path %>
 <%= link_to 'Browse Items', items_path %>
 <%= link_to 'View Merchants', merchants_path %>
+<% if current_admin? %>
```
Comment (wipegup) [inline app/views/shared/_nav.html.erb]: This might be better in the  admin nav partial at: `app/views/shared/_admin_nav.html.erb` -- Which will obviate the need for doing another `current_admin?` check

File: spec/features/admin/merchant_show_spec.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+      @admin = create(:admin)
+      @merchant_1 = create(:merchant)
+      @merchant_2 = create(:merchant)
+      allow_any_instance_of(ApplicationController).to receive(:current_user).and_return(@admin)
+    end
+    it 'shows all the same information that a merchant would see' do
+      visit admin_merchants_path
+
+      within "#merchant-#{@merchant_1.id}" do
+        click_link "#{@merchant_1.name}"
```
Comment (wipegup) [inline spec/features/admin/merchant_show_spec.rb]: Shouldn't need interpolation here, just `click_link @merchant_1.name` should work

File: spec/features/admin/merchant_show_spec.rb [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+      end
+
+      expect(current_path).to eq("/admin/merchants/#{@merchant_1.id}")
```
Comment (wipegup) [inline spec/features/admin/merchant_show_spec.rb]: path helper?

File: spec/features/admin/users_index_spec.rb [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+      allow_any_instance_of(ApplicationController).to receive(:current_user).and_return(@user_1)
+
+      visit root_path
+
+      expect(page).to_not have_link("Users")
+    end
+
+    it 'shows information about each user' do
+
+      visit '/admin/users'
```
Comment (wipegup) [inline spec/features/admin/users_index_spec.rb:29]: path helper?

File: spec/features/admin/users_show_spec.rb [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+      @user_2 = create(:user)
+      @merchant_1 = create(:merchant)
+      allow_any_instance_of(ApplicationController).to receive(:current_user).and_return(@admin)
+    end
+
+    it "Shows all the same information a normal user would see in their profile" do
+      visit admin_users_path
+
+      within "#user-#{@user_1.id}" do
+        click_link "#{@user_1.name}"
```
Comment (wipegup) [inline spec/features/admin/users_show_spec.rb:17]: Can probably do w/o the interpolation here as well

**Review by wipegup — COMMENTED** [review]
A few things that would be nice to clean up, but nothing that should block a merge

File: spec/features/admin/merchant_show_spec.rb (hunk 3 above)
Comment (Vjp888) [reply spec/features/admin/merchant_show_spec.rb]: I was running into an odd issue where I was getting admin/merchants.:id instead of /:id so this was to make sure that my path was routing correctly.

The link itself is uses the path helper though so testing wise it is working correctly.

File: app/views/shared/_nav.html.erb (hunk 1 above)
Comment (whois-jon-peterson) [reply app/views/shared/_nav.html.erb]: Agreed

**Review by whois-jon-peterson — APPROVED** [review]

File: app/views/shared/_nav.html.erb (hunk 1 above)
Comment (wipegup) [reply app/views/shared/_nav.html.erb]: Addressed in new commit
